jax/core.py

```diff
@@ -779,8 +779,11 @@ def full_lower(val):
     return val
 
 def find_top_trace(xs) -> Trace:
-  top_main = max((x._trace.main for x in xs if isinstance(x, Tracer)),
-                 default=None, key=attrgetter('level'))
+  top_tracer, top_main = max(((x, x._trace.main) for x in xs if isinstance(x, Tracer)),
+                             default=(None, None),
+                             key=lambda x: x[1].level)
+  if top_tracer is not None:
+    top_tracer._assert_live()
```

Comment:
happy to put this check somewhere else if there is a better place for it :)

Reply:
This seems like the perfect place!

Riffing on this code, WDYT about writing it like this?

```python
def find_top_trace(xs) -> Trace:
  top_tracer = max((x for x in xs if isinstance(x, Tracer)),
                    default=None, key=attrgetter('_trace.level'))
  if top_tracer is not None:
    top_tracer._assert_live()
    top_main = top_tracer._trace.main
  else:
    top_main = None
  dynamic = thread_local_state.trace_state.trace_stack.dynamic
  top_main = (dynamic if top_main is None or dynamic.level > top_main.level
              else top_main)
  return top_main and top_main.with_cur_sublevel()  # type: ignore
```

Reply:
Done.